csi/crypto/crypto.go

```diff
@@ -85,12 +84,21 @@ func VolumeMapper(volume, dataEngine string) string {
 
 // EncryptVolume encrypts provided device with LUKS.
 func EncryptVolume(devicePath, passphrase string, cryptoParams *EncryptParams) error {
+	isEncrypted, err := IsDeviceEncrypted(devicePath)
+	if err != nil {
```

Comment:
Based on your commit description, there will be an error if the device exists but is not encrypted
```
2. The device is not encrypted -> return exit code 1
```

Hence here do we need to figure out this error by ourselves (and continue the encryption)?

Reply:
Yes, we figure it out and continue here https://github.com/longhorn/longhorn-manager/pull/3566/files#diff-54750e8b890e240ba5a42e35654428da7c3d67323eb4d8961f162b186460bb62R60-R64

Reply:
If not encrypted, IsDeviceEncrypted will return "false, nil", right?

Reply:
> If not encrypted, IsDeviceEncrypted will return "false, nil", right?
Yes, you are correct. Then we proceed to encrypt the volume by `luksFormat` which will wipe out the data